### DISCUSSION TITLE
不可添加，实体没有主键：(0, string)

### DISCUSSION THREAD

**Discussion by wdqzdd** [dotnetcore/FreeSql#1511] [Q&A]
![image](https://user-images.githubusercontent.com/27991992/237059416-0b05a4f3-8f7f-438f-878d-43b47a6c91f3.png)

    public class RepositoryBase<TEntity> : DefaultRepository<TEntity>, IRepositoryBase<TEntity> where TEntity : BaseEntity
    {
        public RepositoryBase(UnitOfWorkManager fsql) : base(fsql.Orm, fsql)
        {

        }
    }

    public class SongDetailRepository : RepositoryBase<SongDetail>, ISongDetailRepository
    {
        public SongDetailRepository(MasterUnitOfWorkManager fsql) : base(fsql)
        {
            
        }
    }

    public class SongDetail: BaseEntity
    {
        public int SongId { get; set; }

        /// <summary>
        /// 标题
        /// https://learn.microsoft.com/zh-cn/aspnet/core/mvc/models/validation?view=aspnetcore-6.0
        /// </summary>
        [StringLength(10, ErrorMessage = "长度不能大于10")]
        public string SongTitleName { get; set; }
    }

![image](https://user-images.githubusercontent.com/27991992/237058844-3a6f9aa4-d2f8-4606-aaec-6328006180ae.png)

**Comment by wdqzdd** [comment]
DefaultRepository《TEntity》是自己定义的。我看到底层有
![image](https://user-images.githubusercontent.com/27991992/237059819-39d8d0fc-c395-462e-82a0-fc5b5cc4981b.png)

**Comment by densen2014** [comment]
你写个console吧

**Reply by wdqzdd** [reply]
[ConsoleApp20.zip](https://github.com/dotnetcore/FreeSql/files/11430219/ConsoleApp20.zip)

**Comment by densen2014** [comment]
Unhandled exception. System.Exception: 不可添加，实体没有主键：(10, jj)
   at FreeSql.DbSet`1.CanAdd(TEntity data, Boolean isThrow)
   at FreeSql.DbSet`1.AddPriv(TEntity data, Boolean isCheck)
   at FreeSql.DbSet`1.Add(TEntity data)
   at FreeSql.BaseRepository`1.Insert(TEntity entity)

**Comment by 2881099** [comment]
仓储，需要依赖实体标记 IsPrimary
